File: sample/src/main/java/com/dazhi/sample/SampleActivity.java
package com.dazhi.sample;

import android.annotation.SuppressLint;
import android.widget.Button;
import android.widget.TextView;
import com.dazhi.libroot.inte.InteCallRoot;
import com.dazhi.libroot.root.RootVmActivity;
import com.dazhi.libroot.util.UtRoot;
import com.dazhi.sample.db.DbPerson;
import java.util.List;
import androidx.annotation.Nullable;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;

public class SampleActivity extends RootVmActivity<VmMain> {

    @Override
    protected int getLayoutId() {
        return R.layout.activity_sample;
    }

    @Override
    protected void initConfig(TextView tvToolTitle) {
        // 初始化ViewModel
        vm = ViewModelProviders.of(this).get(VmMain.class);
        // 设置标题
        if(tvToolTitle!=null){
            tvToolTitle.setText("我的第一个MVVM架构");
        }
    }

    @SuppressLint("CheckResult")
    @Override
    protected void initViewAndDataAndEvent() {
        super.initViewAndDataAndEvent();
        // UI
        final TextView tvMainShowDbData = findViewById(R.id.tvMainShowDbData);
        Button btMainInsertData = findViewById(R.id.btMainInsertData);
        UtRoot.rxViewClick(btMainInsertData, new InteCallRoot(){
            @Override
            public void call() {
                vm.insertLsBnPerson();
            }
        });
        Button btMainDeleteData = findViewById(R.id.btMainDeleteData);
        UtRoot.rxViewClick(btMainDeleteData, new InteCallRoot(){
            @Override
            public void call() {
                String str=tvMainShowDbData.getText().toString();
//                List<DbPerson> lsBn=Arrays.asList(new Gson().fromJson(str, DbPerson[].class));
//                lsBn.remove(0);
//                List<BnPerson> lsBn= new ArrayList<>();
//                BnPerson bn=new BnPerson();
//                bn.setId(118L);
//                lsBn.add(bn);
//                vm.deleteLsBnPerson(lsBn);
            }
        });

        // 数据变动监听
        vm.getDbLsBnPerson().observe(this, new Observer<List<DbPerson>>() {
            @Override
            public void onChanged(@Nullable List<DbPerson> lsBnPeople) {
//                String strTemp=new Gson().toJson(lsBnPeople);
//                tvMainShowDbData.setText(strTemp);
            }
        });
    }


}

File: sample/src/main/java/com/dazhi/sample/SampleActivity.kt
package com.dazhi.sample

import android.annotation.SuppressLint
import android.content.Intent
import android.widget.TextView
import androidx.lifecycle.ViewModelProvider
import com.dazhi.libroot.root.RootVmActivity
import com.dazhi.libroot.util.RtCmn
import com.dazhi.libroot.util.viewClick
import com.dazhi.sample.databinding.ActivitySampleBinding

class SampleActivity : RootVmActivity<VmMain, ActivitySampleBinding>() {

    override fun initBinding(): ActivitySampleBinding {
        return ActivitySampleBinding.inflate(layoutInflater)
    }

    override fun initVm(): VmMain {
        return ViewModelProvider(this).get(VmMain::class.java)
    }

    override fun initConfig(tvToolTitle: TextView?) {
        super.initConfig(tvToolTitle)
        // 设置标题
        tvToolTitle?.text = "我的第一个MVVM架构"
    }

    @SuppressLint("CheckResult")
    override fun initViewAndDataAndEvent() {
        super.initViewAndDataAndEvent()
        // UI
//        var lastTime=System.currentTimeMillis()
        viewClick(binding.btMainInsertData) {
            // 测试防抖动
//            val temp = System.currentTimeMillis()
//            RtLog.d("间隔：${temp-lastTime}")
//            lastTime = temp
//            return@viewClick
            //
            RtCmn.toastShort("测试插入")
            vm.insertLsBnPerson()
        }

        viewClick(binding.btMainDeleteData) {
//            binding.tvMainShowDbData.text.toString()
//            val lsBnPerson: MutableList<DbPerson> = ArrayList()
//            for (i in 1..10) {
//                lsBnPerson.add(DbPerson("name0$i"))
//            }
//            vm.deleteLsBnPerson(lsBnPerson);
        }
        // 数据变动监听
//        vm.dbLsBnPerson.observe(this) { mList ->
//            binding.tvMainShowDbData.setText(mList.map {
//                "名称：${it.strName}"
//            }.toString());
//        }

        // 跳转实例
        viewClick(binding.btMainGotoNewPage) {
            startActivity(Intent(this, SimpleActivity::class.java))
        }
    }

}